### ISSUE TITLE
Upgrade *DataContext*

### ISSUE THREAD

**Issue by vCrespoP** [Alecrim/AlecrimCoreData#157]
Hello, I'm having some trouble migrating to the latest version but I think is totally worth.

I have some inherited code which relies on "DataContext" and I was wondering how can I upgrade the syntax or the classes that I may need to use. For example:

```
let documentsUrl = FileManager.default.urls(for: .documentDirectory, in:.userDomainMask).last!
        let databaseUrl = documentsUrl.appendingPathComponent("MYDB.db")
        let modelURL = Bundle.main.url(forResource: "MYMODEL", withExtension: "momd")!

        if let dataContextOptions = try? DataContextOptions(managedObjectModelURL: modelURL, persistentStoreURL: databaseUrl) {
            dataContext = DataContext(dataContextOptions: dataContextOptions)
        } else {
            dataContext = DataContext()
        }
```

Any ideas how should I proceed?

Thank you very much for your time! :D

**Comment by vCrespoP** [comment]
Thanks for your @vmartinelli it sure does help a lot!!

Strange thing is I'm getting the terrible "Segmentation fault: 11" in the class 
MyCustomPersistentContainer you described

I've tried resetting Xcode, removing derived data and the usual things without success. Is this compiling ok for you? It seems the syntax is ok :/

Thanks again and sorry for the inconvenience.

**Comment by vCrespoP** [comment]
I got it!

If you use it like this:
```
super.init(name: "MYDB", managedObjectModel: model, automaticallyLoadPersistentStores: true)
```

It doesn't fail anymore, so maybe there's some issue around that initializer without the last parameter you posted in the answer?

**Comment by vmartinelli** [comment]
@vCrespoP I think you catched a Swift linker error. I'll do more tests regarding this issue. About the autocomplete, no, it is not normal.

**Comment by vmartinelli** [comment]
Fixed in release 5.0.
